### PR TITLE
Added parallelized travis builds and tests

### PR DESCRIPTION
**Pull request by Snaipe**
I went and fixed any address clashes in the tests to make them run in parallel.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -211,12 +211,14 @@ if (NN_TESTS)
...
 
+    set (TEST_PORT 5500)
     macro (add_libnanomsg_test NAME TIMEOUT)
         list (APPEND all_tests ${NAME})
         add_executable (${NAME} tests/${NAME}.c)
         target_link_libraries (${NAME} ${PROJECT_NAME})
-        add_test (NAME ${NAME} COMMAND ${NAME})
+        add_test (NAME ${NAME} COMMAND ${NAME} ${TEST_PORT})
         set_tests_properties (${NAME} PROPERTIES TIMEOUT ${TIMEOUT})
+        math (EXPR TEST_PORT "${TEST_PORT}+10")
```
Comment (gdamore) [inline CMakeLists.txt]: Brilliant!

**Comment by gdamore** [comment]
I'm pretty well pleased by this.  You should add copyrights to the files modified, and then make sure this is formatted properly with the commit comment (file a bug "tests should run in parallel") and make the commit comment "fixes #<number> tests should run in parallel"

Once that's done, we can merge.  Thanks for the contribution!

**Comment by gdamore** [comment]
Whoops.  The PR failed to build ... the ws test has compile time errors:

/home/travis/build/nanomsg/nanomsg/tests/ws.c:85:27: error: ‘argc’ undeclared (first use in this function)
/home/travis/build/nanomsg/nanomsg/tests/ws.c:85:27: note: each undeclared identifier is reported only once for each function it appears in
/home/travis/build/nanomsg/nanomsg/tests/ws.c:85:33: error: ‘argv’ undeclared (first use in this function)

Please fix and resubmit.

**Comment by JackDunaway** [comment]
Cannot stress the value introduced with this commit. Introducing chaos to these tests should flush out all manners of parallel/concurrency issues.

Does CMake also offer the opportunity to specify number of iterations per test? And also, does it offer ability to randomize test order?

**Comment by Snaipe** [comment]
@JackDunaway there's `--repeat-until-fail <n>` to repeat each test n times, and `--schedule-random` to randomize them.

**Pull request merged**